src/Compilers/Core/Portable/Syntax/GreenNode.cs

```diff
@@ -260,116 +250,140 @@ public virtual int FindSlotIndexContainingOffset(int offset)
         internal enum NodeFlags : ushort
         {
             None = 0,
-            ContainsDiagnostics = 1 << 0,
-            ContainsStructuredTrivia = 1 << 1,
-            ContainsDirectives = 1 << 2,
-            ContainsSkippedText = 1 << 3,
-            ContainsAnnotations = 1 << 4,
-            IsNotMissing = 1 << 5,
-            ContainsAttributes = 1 << 6,
-
-            FactoryContextIsInAsync = 1 << 7,
-            FactoryContextIsInQuery = 1 << 8,
+            /// <summary>
+            /// If this node is missing or not.  We use a non-zero value for the not-missing case so that this value
+            /// automatically merges upwards when building parent nodes.  In other words, once we have one node that is
+            /// not-missing, all nodes above it are definitely not-missing as well.
+            /// </summary>
+            IsNotMissing = 1 << 0,
+            /// <summary>
+            /// If this node directly has annotations (not its descendants).  <see cref="ContainsAnnotations"/> can be
+            /// used to determine if a node or any of its descendants has annotations.
+            /// </summary>
+            HasAnnotationsDirectly = 1 << 1,
+
+            FactoryContextIsInAsync = 1 << 2,
+            FactoryContextIsInQuery = 1 << 3,
             FactoryContextIsInIterator = FactoryContextIsInQuery,  // VB does not use "InQuery", but uses "InIterator" instead
 
-            InheritMask = ContainsDiagnostics | ContainsStructuredTrivia | ContainsDirectives | ContainsSkippedText | ContainsAnnotations | ContainsAttributes | IsNotMissing,
-        }
+            // Flags that are inherited upwards when building parent nodes.  They should all start with "Contains" to
+            // indicate that the information could be found on it or anywhere in its children.
 
-        internal NodeFlags Flags
-        {
-            get { return this.flags; }
+            /// <summary>
+            /// If this node, or any of its descendants has annotations attached to them.
+            /// </summary>
+            ContainsAnnotations = 1 << 4,
+            /// <summary>
+            /// If this node, or any of its descendants has attributes attached to it.
+            /// </summary>
+            ContainsAttributes = 1 << 5,
+            ContainsDiagnostics = 1 << 6,
+            ContainsDirectives = 1 << 7,
+            ContainsSkippedText = 1 << 8,
+            ContainsStructuredTrivia = 1 << 9,
+
+            InheritMask = IsNotMissing | ContainsAnnotations | ContainsAttributes | ContainsDiagnostics | ContainsDirectives | ContainsSkippedText | ContainsStructuredTrivia,
         }
 
         internal void SetFlags(NodeFlags flags)
```

Comment:
note: i wanted to make these protected.  That works in C#, but vb has a few places taht call these directly on a node (odd).  

this is also nice as we now only expose the ability to set/clear bits.  not the ability to accidentally assign over the node-flags entirely.